rp2040-hal/src/pio.rs

```diff
@@ -292,7 +292,25 @@ pub struct InstalledProgram<P> {
 }
 
 impl<P: PIOExt> InstalledProgram<P> {
-    /// Get the warp target (entry point) of the instaled program.
+    /// Change the source and/or target for automatic program wrapping.
+    ///
+    /// This replaces the current wrap bounds with a new set. This can be useful if you are running
+    /// multiple state machines with the same program but using different wrap bounds.
+    ///
+    /// # Returns
+    ///
+    /// * [`Ok`] containing a new program with the provided wrap bounds
+    /// * [`Err`] containing the old program if the provided wrap was invalid (outside the bounds of
+    ///   the program length)
+    pub fn set_wrap(self, wrap: Wrap) -> Result<Self, Self> {
+        if (self.offset + wrap.source) < self.length && (self.offset + wrap.target) < self.length {
```

Comment:
Why is the offset relevant in those checks?
Aren't wrap.source and wrap.target relative to the offset, the same way length is?
So both source and target should be between 0 and length-1, inclusive.

Reply:
Thanks, I misunderstood how that worked. Changed now